src/bio/bio_xstream.c

```diff
@@ -892,7 +891,34 @@ create_bio_bdev(struct bio_xs_context *ctxt, const char *bdev_name,
 }
 
 static int
-init_bio_bdevs(struct bio_xs_context *ctxt)
+bdev_name2roles(const char *name)
+{
+	const char	*dst = strrchr(name, '_');
+	char		*ptr_parse_end = NULL;
+	unsigned	 int value;
+
+	if (dst == NULL)
+		return -DER_NONEXIST;
+
+	dst++;
+	value = strtoul(dst, &ptr_parse_end, 0);
+	if (ptr_parse_end && *ptr_parse_end != '\0') {
+		D_ERROR("invalid numeric value: %s (name %s)\n", dst, name);
+		return -DER_INVAL;
+	}
+
+	if (value & (~NVME_ROLE_ALL))
+		return -DER_INVAL;
+
+	/* AIO */
+	if (value == 0)
+		value = NVME_ROLE_ALL;
```

Comment:
This is temporary workaround for testing? If so, it needs be removed before landing.